File: investigation/kotlin_probe/main_module/src/kz/greetgo/kotline_probe/main/launcher/MainLauncher.java
package kz.greetgo.kotline_probe.main.launcher;

import kz.greetgo.depinject.Depinject;
import kz.greetgo.kotline_probe.main.beans.MainBean;

public class MainLauncher {
  public static void main(String[] args) {
    new MainLauncher().launch();
  }

  private void launch() {
    BeanContainerMain beanContainer = Depinject.newInstance(BeanContainerMain.class);

    MainBean mainBean = beanContainer.main();

    mainBean.printHelloWorld();
  }
}

File: investigation/kotlin_probe/main_module/src/kz/greetgo/kotlin_probe/main/launcher/MainLauncher.kt
package kz.greetgo.kotlin_probe.main.launcher

import kz.greetgo.depinject.Depinject
import kz.greetgo.kotlin_probe.main.beans.MainBean

class MainLauncher {

  companion object {
    @JvmStatic
    fun main(args: Array<String>) {
      MainLauncher().launch()
    }
  }

  private fun launch() {
    val beanContainer = Depinject.newInstance(BeanContainerMain::class.java)

    val mainBean = beanContainer.main()

    mainBean.printHelloWorld()
  }
}
